src/android/shakeDetector.java

```diff
@@ -0,0 +1,251 @@
+                    int mutedIndex = findEntryWithId(mutedConfigs, currNotifyConfig.getLong(ID));
+                    if(mutedIndex == -1) {
+                        if (autogenData != null) { // we need to merge in the autogenerated data with any user data
+                            JSONObject currData = currNotifyConfig.optJSONObject("data");
+                            if (currData == null) {
+                                currData = new JSONObject();
+                                currNotifyConfig.put("data", currData);
+                            }
+                            mergeObjects(currData, autogenData);
+                        }
+                        Log.d(context, TAG, "generating notification for event "+eventName
+                                + " and id = " + currNotifyConfig.getLong(ID));
+                        Manager.getInstance(context).schedule(currNotifyConfig, TriggerReceiver.class);
+                    } else {
+                        Log.d(context, TAG, "notification for event "+eventName+" and id = "+currNotifyConfig.getLong(ID)
+                                +" muted, skip");
+                    }
+                } catch (Exception e) {
+                    Log.e(context, TAG, "Got error "+e.getMessage()+" while processing object "
+                            + notifyConfigs.getJSONObject(i) + " at index "+i);
+                }
+            }
+        } catch(JSONException e) {
+            Log.e(context, TAG, e.getMessage());
+            Log.e(context, TAG, e.toString());
+        }
+    }
+
+    private void mergeObjects(JSONObject existing, JSONObject autogen) throws JSONException {
+        JSONArray toBeCopiedKeys = autogen.names();
+        for(int j = 0; j < toBeCopiedKeys.length(); j++) {
+            String currKey = toBeCopiedKeys.getString(j);
+            existing.put(currKey, autogen.get(currKey));
+        }
+    }
+
+    private int findEntryWithId(JSONArray array, long id) throws JSONException {
+        if (array == null) {
+            return -1;
+        }
+        for (int i = 0; i < array.length(); i++) {
+            JSONObject currCheckedObject = array.getJSONObject(i);
+            if (currCheckedObject.getLong(ID) == id) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    private void storeDataAndNotify(Context context, float[] accelerations, long currTime, float combinedAccel) throws JSONException{
+
+        JSONObject incidentData = new JSONObject();
+        UserCache uc = UserCacheFactory.getUserCache(this);
+        SimpleLocation loc[] = uc.getLastSensorData(R.string.key_usercache_filtered_location, 1, SimpleLocation.class);
+
+        if (loc.length == 0) {
+            incidentData.put("inc_ts", ((double) currTime) / 1000);
+        } else {
+            SimpleLocation mostRecentLocation = loc[0];
+            incidentData.put("inc_ts", ((double) currTime) / 1000);
```

Comment:
Instead of copying every single entry over, you should have this be a field `"inc"` whose value is the JSON representation of the SimpleLocation, as serialized by Gson (e.g. `new Gson().toJson(cfg))`). See https://github.com/google/gson for more details.